txnsync/outgoing_test.go

```diff
@@ -0,0 +1,381 @@
+
+	expectedFilter := s.makeBloomFilter(requestParams{Offset: 111, Modulator: 222}, pendingTransactions.pendingTransactionsGroups, &s.lastBloomFilter)
+
+	s.isRelay = true
+	peer.isOutgoing = true
+	peer.state = peerStateLateBloom
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
+
+	a.Equal(metaMessage.message.UpdatedRequestParams.Modulator, byte(222))
+	a.Equal(metaMessage.message.UpdatedRequestParams.Offset, byte(111))
+	a.Equal(metaMessage.peer, &peer)
+	a.Equal(metaMessage.message.Version, int32(txnBlockMessageVersion))
+	a.Equal(metaMessage.message.Round, s.round)
+	a.True(metaMessage.message.MsgSync.ResponseElapsedTime != 0)
+	a.Equal(s.lastBloomFilter, expectedFilter)
+}
+
+// TestAssemblePeerMessage2 Tests assemblePeerMessage with messageConstNextMinDelay | messageConstUpdateRequestParams msgOps
+func TestAssemblePeerMessage2(t *testing.T) {
+
+	a := require.New(t)
+
+	s := syncState{clock: timers.MakeMonotonicClock(time.Now())}
+
+	s.profiler = makeProfiler(1*time.Millisecond, s.clock, s.log, 1*time.Millisecond)
+
+	peer := Peer{}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+
+	s.isRelay = true
+	s.lastBeta = 123 * time.Nanosecond
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
+
+	a.Equal(metaMessage.message.UpdatedRequestParams.Modulator, byte(222))
+	a.Equal(metaMessage.message.UpdatedRequestParams.Offset, byte(111))
+	a.Equal(metaMessage.peer, &peer)
+	a.Equal(metaMessage.message.Version, int32(txnBlockMessageVersion))
+	a.Equal(metaMessage.message.Round, s.round)
+	a.True(metaMessage.message.MsgSync.ResponseElapsedTime != 0)
+	a.Equal(metaMessage.message.MsgSync.NextMsgMinDelay, uint64(s.lastBeta.Nanoseconds())*2)
+
+}
+
+// TestAssemblePeerMessage3 Tests assemblePeerMessage messageConstTransactions msgOps
+func TestAssemblePeerMessage3(t *testing.T) {
+	a := require.New(t)
+
+	s := syncState{clock: timers.MakeMonotonicClock(time.Now())}
+
+	s.profiler = makeProfiler(1*time.Millisecond, s.clock, s.log, 1*time.Millisecond)
+
+	peer := Peer{}
+
+	pendingTransactions := pendingTransactionGroupsSnapshot{
+		latestLocallyOriginatedGroupCounter: 1,
+		pendingTransactionsGroups: []transactions.SignedTxGroup{
+			transactions.SignedTxGroup{
+				LocallyOriginated: true,
+				EncodedLength:     2,
+			},
+		},
+	}
+
+	peer.setLocalRequestParams(111, 222)
+	peer.lastReceivedMessageTimestamp = 100
+	peer.lastReceivedMessageLocalRound = s.round
+	peer.requestedTransactionsModulator = 2
+	peer.recentSentTransactions = makeTransactionCache(5, 10, 20)
+
+	s.isRelay = false
+	peer.isOutgoing = true
+	peer.state = peerStateHoldsoff
+
+	metaMessage := s.assemblePeerMessage(&peer, &pendingTransactions)
+
+	a.Equal(len(metaMessage.transactionGroups), 1)
+	a.True(reflect.DeepEqual(metaMessage.transactionGroups[0], pendingTransactions.pendingTransactionsGroups[0]))
```

Comment:
do we need to check here that other fields of metaMessage should be empty?

Reply:
Additionally, I see you use reflect.DeepEqual quite often. Is `require.Equal` not sufficient? I could be wrong here.

Reply:
I prefer DeepEqual only because it will dereference any pointers to point the values itself...open to other suggestions however.